File: js-basics/data-structures/stack.py

class Ray_Stack:
    def __init__(self):
        self.stack = []
        
    def push(self, object: any):
        self.stack.append(object)
        
    def pop(self):
        try:
            var = self.stack[-1]
            self.stack = self.stack[:-1]
            return var
        except:
            raise IndexError

    def isEmpty(self):
        if self.stack:
            return False
        else:
            return True
        
    def getSize(self):
        return len(self.stack)
    
    def peek(self):
        return self.stack[-1]
    

File: js-basics/data-structures/stack.js

class Ray_Stack {
    constructor() {
        this.stack = []
    }
    push(data) {
        this.stack.push(data);
    }
    pop() {return this.stack.pop()}
    
    peek() {
        return this.stack[-1];
    }
    is_empty() {
        if (this.stack.length > 0) {return false}
        else {return true}
    }
    get_size() {return this.stack.length}
}

const stack = new Ray_Stack();
stack.push('1')
console.log(stack.pop())